docs/temporal.md

```diff
@@ -0,0 +1,229 @@
+# Durable Execution with Temporal
+
+Pydantic AI allows you to build durable agents that never lose their progress and handle long-running, asynchronous, and human-in-the-loop workflows with production-grade reliability. Durable agents have full support for [streaming](agents.md#streaming-all-events) and [MCP](mcp/client.md), with the added benefit of fault tolerance.
+
+[Temporal](https://temporal.io) is a popular [durable execution](https://docs.temporal.io/evaluate/understanding-temporal#durable-execution) platform that's natively supported by Pydantic AI.
+The integration only uses Pydantic AI's public interface, so it can also serve as a reference for how to integrate with another durable execution systems.
+
+### Durable Execution
+
+In Temporal's durable execution implementation, a program that crashes or encounters an exception while interacting with a model or API will retry until it can successfully complete.
+
+Temporal relies primarily on a replay mechanism to recover from failures.
+As the program makes progress, Temporal saves key inputs and decisions, allowing a re-started program to pick up right where it left off.
+
+The key to making this work is to separate the applications repeatable (deterministic) and non-repeatable (non-deterministic) parts:
+
+1. Deterministic pieces, termed [**workflows**](https://docs.temporal.io/workflow-definition), execute the same way when re-run with the same inputs.
+2. Non-deterministic pieces, termed [**activities**](https://docs.temporal.io/activities), can run arbitrary code, performing I/O and any other operations.
+
+Workflow code can run for extended periods and, if interrupted, resume exactly where it left off.
```

Comment:
I don't think you explicitly mention anywhere that workflows cannot incorporate I/O. This might be implicit in saying "deterministic" but I feel like when most people read "deterministic" they think non-random, rather than not-including-I/O. I think it's worth making the point explicitly:
```suggestion
Workflow code can run for extended periods and, if interrupted, resume exactly where it left off. Critically, workflow code _cannot_ include any kind of I/O, over network, terminal, etc.
```
or similar